packages/react/menu/src/Menu.tsx

```diff
@@ -94,65 +211,157 @@ const MenuContent = React.forwardRef((props, forwardedRef) => {
       <CollectionSlot>
-        <MenuContentImpl
-          data-state={getOpenState(context.open)}
-          {...contentProps}
-          ref={forwardedRef}
-        />
+        {context.isSubmenu ? (
+          <MenuSubContent {...contentProps} ref={forwardedRef} />
+        ) : (
+          <MenuRootContent {...contentProps} ref={forwardedRef} />
+        )}
       </CollectionSlot>
     </Presence>
   );
 }) as MenuContentPrimitive;
 
-type FocusScopeOwnProps = Polymorphic.OwnProps<typeof FocusScope>;
-type DismissableLayerOwnProps = Polymorphic.OwnProps<typeof DismissableLayer>;
-type RovingFocusGroupOwnProps = Polymorphic.OwnProps<typeof RovingFocusGroup>;
+/* ---------------------------------------------------------------------------------------------- */
 
-type MenuContentImplOwnProps = Polymorphic.Merge<
-  Polymorphic.OwnProps<typeof PopperPrimitive.Content>,
-  Omit<DismissableLayerOwnProps, 'onDismiss'> & {
-    /**
-     * Whether focus should be trapped within the `MenuContent`
-     * (default: false)
-     */
-    trapFocus?: FocusScopeOwnProps['trapped'];
+type MenuRootContentOwnProps = Omit<
+  Polymorphic.OwnProps<typeof MenuContentImpl>,
+  keyof MenuContentImplPrivateProps
+>;
+type MenuRootContentPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof MenuContentImpl>,
+  MenuRootContentOwnProps
+>;
 
-    /**
-     * Event handler called when auto-focusing on open.
-     * Can be prevented.
-     */
-    onOpenAutoFocus?: FocusScopeOwnProps['onMountAutoFocus'];
+const MenuRootContent = React.forwardRef((props, forwardedRef) => {
+  const context = useMenuContext(CONTENT_NAME);
+  const ref = React.useRef<React.ElementRef<typeof MenuContentImpl>>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
 
-    /**
-     * Event handler called when auto-focusing on close.
-     * Can be prevented.
-     */
-    onCloseAutoFocus?: FocusScopeOwnProps['onUnmountAutoFocus'];
+  // Hide everything from ARIA except the `MenuContent`
+  React.useEffect(() => {
+    const content = ref.current;
+    if (content) return hideOthers(content);
+  }, []);
 
-    /**
-     * Whether scrolling outside the `MenuContent` should be prevented
-     * (default: `false`)
-     */
-    disableOutsideScroll?: boolean;
+  return (
+    <MenuContentImpl {...props} ref={composedRefs} onDismiss={() => context.onOpenChange(false)} />
+  );
+}) as MenuRootContentPrimitive;
 
-    /**
-     * The direction of navigation between menu items.
-     * @defaultValue ltr
-     */
-    dir?: RovingFocusGroupOwnProps['dir'];
+/* ---------------------------------------------------------------------------------------------- */
 
-    /**
-     * Whether keyboard navigation should loop around
-     * @defaultValue false
-     */
-    loop?: RovingFocusGroupOwnProps['loop'];
+type MenuSubContentOwnProps = Omit<
+  Polymorphic.OwnProps<typeof MenuContentImpl>,
+  | keyof MenuContentImplPrivateProps
+  | 'align'
```

Comment:
Small detail but are we sure we should be omitting `align` here? this should be user adjustable for submenus right?

Reply:
No I think that's correct, because we hardcode `start` a bit below as you can see.
That's because that's how sub-menus should be positioned, that said it's also temporary until we redo more specific positioning for menus in general.

But yeah all in all that's correct I think.